tests/serverpod_cli_e2e_test/test/generate_watch_test.dart

```diff
@@ -371,6 +371,188 @@ class TestEndpoint extends Endpoint {
+      () {
+    var (projectName, commandRoot) = createRandomProjectName(tempPath);
+    var (serverDir, _, _) = createProjectFolderPaths(projectName);
+
+    late Process createProcess;
+    Process? generateWatch;
+    KeywordSearchInStream generateStreamSearch = KeywordSearchInStream(
+      keywords: generateWatchCompletionKeywords,
+    );
+    setUp(() async {
+      // Create project
+      createProcess = await Process.start(
+        'serverpod',
+        ['create', projectName, '-v', '--no-analytics'],
+        workingDirectory: tempPath,
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      createProcess.stdout.transform(const Utf8Decoder()).listen(print);
+      createProcess.stderr.transform(const Utf8Decoder()).listen(print);
+
+      var createProjectExitCode = await createProcess.exitCode;
+      assert(
+        createProjectExitCode == 0,
+        'Failed to create the serverpod project.',
+      );
+    });
+
+    tearDown(() async {
+      createProcess.kill();
+      generateWatch?.kill();
+      generateStreamSearch.cancel();
+
+      await Process.run(
+        'docker',
+        ['compose', 'down', '-v'],
+        workingDirectory: commandRoot,
+      );
+
+      while (!await isNetworkPortAvailable(8090)) {}
+    });
+
+    test('then endpoint dispatcher is generated and updated as expected.',
+        () async {
+      // Start generate watch
+      generateWatch = await Process.start(
+        'serverpod',
+        ['generate', '--watch', '-v', '--no-analytics'],
+        workingDirectory: path.join(tempPath, serverDir),
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      generateWatch!.stdout
+          .transform(const Utf8Decoder())
+          .transform(const LineSplitter())
+          .listen(generateStreamSearch.onData);
+      generateWatch!.stderr
+          .transform(const Utf8Decoder())
+          .transform(const LineSplitter())
+          .listen(print);
+
+      await expectLater(
+        generateStreamSearch.keywordFound,
+        completion(isTrue),
+        reason:
+            'Initial code generation did not complete before timeout was reached.',
+      );
+      // This delay is required to ensure that the generate watch is
+      // ready to receive file changes after the initial generation.
+      await Future.delayed(const Duration(seconds: 1));
+
+      // Add endpoint file
+      var endpointFile = File(createProjectDartFilePath(
+        tmpFolder: tempPath,
+        serverDir: serverDir,
+        fileName: 'test_endpoint',
+      ));
+      endpointFile.createSync(recursive: true);
+      endpointFile.writeAsStringSync('''
+import 'package:serverpod/serverpod.dart';
+
+class TestEndpoint extends Endpoint {
+  Future<String> testEndpointMethod(Session session, String name) async {
+    return 'Hello \$name';
+  }
+}
+
+''', flush: true);
+
+      await expectLater(
+        generateStreamSearch.keywordFound,
```

Comment:
Question: How does this work?

Reply:
This is simply a future that watches the stdout stream for keywords, once it is found the future completes.

Reply:
Alright, I kind of think it would be nice to see what we are looking for here, I couldn't even see these in any changes you did, so I assume they are defined at the top=

Reply:
Yes, correct. If I were to reimplement these, they should probably be made more explicit with a list of what is expected at each stage. The tooling for these tests need some help. I might go back and redo this at a later stage.